### PR TITLE
feat: add primary key support

### PR DESCRIPTION
**Pull request by TopKeyboard**

### CODE REVIEW

File: engine/db/table_segment_mvp.cpp [hunk 1]
```diff
@@ -306,9 +393,17 @@ Status TableSegmentMVP::Insert(meta::TableSchema& table_schema, Json& records, i
       }
     }
     ++cursor;
+
+  LOOP_END : {}
+    // nothing should be done at the end of block
   }
   record_number_.store(cursor);
-  return Status::OK();
+  auto msg = std::string("loading done. added entries: " +
```
Comment (richard-epsilla) [inline engine/db/table_segment_mvp.cpp:401]: auto msg = "Successfully inserted " + std::to_string(new_record_size - skipped_entry) + records."
if (skipped_entry > 0) {
  msg += "Skipped " + std::to_string(skipped_entry) + " records with primary key values that already exist."
}

Comment (richard-epsilla) [reply engine/db/table_segment_mvp.cpp:401]: Also, need change WebController to respond the message back to end user

File: engine/db/table_segment_mvp.cpp [hunk 2]
```diff
@@ -52,6 +52,10 @@ Status TableSegmentMVP::Init(meta::TableSchema& table_schema, int64_t size_limit
     if (field_schema.field_type_ == meta::FieldType::STRING) {
       field_id_mem_offset_map_[field_schema.id_] = string_num_;
       field_name_mem_offset_map_[field_schema.name_] = string_num_;
+      if (field_schema.is_primary_key_) {
+        assert(!primitive_pk_field_id_);  // only one pk allowed
```
[the remark this answers is not in the document]
Comment (TopKeyboard) [reply engine/db/table_segment_mvp.cpp]: already have check in `/home/vectordb/vectordb/engine/db/catalog/basic_meta_impl.cpp` 
removed the assert in 
1910e899dab4f06e3ab7f4ee4d58f3c352d8c239

File: engine/db/table_segment_mvp.cpp (hunk 1 above)
Comment (TopKeyboard) [reply engine/db/table_segment_mvp.cpp]: messaged improved in e9d07c260c76b57a4e6109815faaada478989bdd
web controller response added in 7be4c52d5e7118fc47c6339c94873080e3fdc95e

**Review by richard-epsilla — APPROVED** [review]

**Pull request merged**